### PR TITLE
DART: new tree booster provides Dropout

### PR DESCRIPTION
**Pull request by marugari**
Is it OK?
https://github.com/dmlc/xgboost/issues/809#issuecomment-228819113

### CODE REVIEW

File: _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+comments: true
+---
+
+## Introduction
+Gradient Boosting Trees algorithms(especially [XGBoost](https://github.com/dmlc/xgboost)) give awesome performance in lots of [competitions](https://github.com/dmlc/xgboost/tree/master/demo#machine-learning-challenge-winning-solutions).
+They combine a huge number of regression trees with small learning rate.
+
+In this situation, trees added early are significance and trees added late are unimportant.
+DART algorithm drops trees added earlier to level contributions.
+
```
Comment (tqchen) [inline _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown]: Recently, Rasmi et.al proposed a new method to add dropout techniques from deep neural nets community to trees, and reported better results in some situations. We are glad to announce that DART is now supported in XGBoost, taking fully benefit of all xgboost optimizations to this new technique

File: _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown [hunk 2]
```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: "DART: new tree booster provides Dropout"
```
Comment (tqchen) [inline _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown]: Change to "Support Dropout on XGBoost"

**Comment by tqchen** [comment]
made a few comments, otherwise it looks nice.

File: _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Comparing `dart` with `gbtree` using the dataset from [Santander Customer Satisfaction](https://www.kaggle.com/c/santander-customer-satisfaction).
+
+In this experiment, `dart` shows better classification performance.
+
+![alt text](https://raw.githubusercontent.com/dmlc/web-data/master/xgboost/dart/dart_experiment.png)
+
+The code is [here (external link)](https://github.com/marugari/Notebooks/blob/master/test_dart.ipynb).
+
+## Bio
+Yoshinori Nakano works as a data analyst in the field of credit scoring.
```
Comment (tqchen) [inline _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown]: This feature is contributed by [Yoshinori Nakano](your homepage) .

File: _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+## Original paper
+Rashmi Korlakai Vinayak, Ran Gilad-Bachrach. "DART: Dropouts meet Multiple Additive Regression Trees."
+
+[JMLR](http://www.jmlr.org/proceedings/papers/v38/korlakaivinayak15.pdf)
+
+## Features
+- Drop trees in order to solve the over-fitting.
+  - Trivial trees (to correct trivial errors) may be prevented.
+- Slower than `gbtree`.
```
Comment (tqchen) [inline _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown]: Note that due to the randomness in dropout. Currently dart runs slower than gbtree, and cannot directly benefit from early stopping(because evaluation metric fluctuates).

File: _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown (hunk 3 above)
Comment (tqchen) [inline _posts/2016-07-03-dart-new-tree-booster-provides-dropout.markdown]: add link to your homepage :)

**Comment by marugari** [comment]
Thank you for changing what I forget to change.
